### DISCUSSION TITLE
Cannot connect to open sessions and running programs, I can see them running in the background

### DISCUSSION THREAD

**Discussion by liuyanghu** [neutrinolabs/xrdp#2108] [Q&A] [answered]
I have a problem. When I use a different computer to connect to Ubuntu via xrdp every time, I cannot connect to the open sessions and running programs. Work and procedures are also lost. So I want to ask, can xrdp restrict all connections of the same user to the same session？

Maybe someone else has raised this question, if so please let me know, I'm happy to study it myself

**Comment by matt335672** [comment, marked as answer]
@liuyanghu - I don't understand exactly what you are trying to do.

If you want to connect to a session on the machine console you will need additional software. Is this what you are trying to do?

If you're trying to connect to an existing xrdp session, the behaviour will depend on your version of xrdp and the backend you are using. Different versions of Ubuntu use different versions of xrdp. What is your version of xrdp as displayed by `xrdp -v`?

**Reply by liuyanghu** [reply]
Maybe my words are a little confusing. I use an example to illustrate my problem. There are two Windows 10 computers, mine (abbreviated as A) and my colleague (abbreviated as B). A Ubuntu linux server with xrdp 0.9 and MATE desktop environment installed.

 I connected ubuntu via Windows mstsc on computer A, opened Matlab or Spyder, and ran a program. 
Then I connect to the **same user** via mstsc on computer B **again** (emphasis: still my own ubuntu account), **xrdp will open a new session for me but I can’t see MATLAB and spyder.**

 I can see them through the linux top command. **They are still running in the background**, but I can no longer see them on the desktop.
Can xrdp restrict all connections of the **same user** to the same session? This is not a common phenomenon, but it still happens occasionally. Then the software and running programs I opened before can no longer be operated, and I can't see the results, even though they are still running in the background. This caused me a lot of trouble.

Best wishes

**Reply by matt335672** [reply]
I understand perfectly now - that's a great description.

I have two questions for you:-
1) What backend are you selecting in the top dropdown on the login screen? Is it Xvnc or Xorg?
2) What is the output of `xrdp -v`?

With that information I should be able to explain to you what is happening and how you can connect to the same session from both A and B.

Thanks.

**Reply by liuyanghu** [reply]
Thanks for reply。xrdp -v returns 0.6.1。login screen：sesman-Xvnc

**Reply by matt335672** [reply]
OK - that explains a lot. That's a very old version of xrdp. You must be running Ubuntu 16.04.

With that version of xrdp and sesman-Xvnc, the session is not resizeable. So when user A starts the session from his client, the session will have a certain width and height (say width = 1280, height = 768) and a certain colour depth (say 24 bits).

For user B to connect to the same session on this version, all of these three items must match. So, in this example, that means that user B will have to select a screen size of 1280x768, and a colour depth of 24 on the client parameters before connecting. Here is a screenshot showing what I mean:-

![Capture](https://user-images.githubusercontent.com/30179339/148392393-f01a4b2d-816f-4d78-8c17-40e60298abc2.PNG)

Does that make sense?

Later versions of xrdp support this a lot better, but even so, this method will work with them too.

**Reply by liuyanghu** [reply]
Thank you very much, your answer explained my confusion. It is also time for IT administrators to upgrade the version (haha). Thank you for your time and answers.

**Reply by matt335672** [reply]
One other thing - if you can't find the width and height values you need in the mstsc screen above, you can always edit the connection ini file with notepad and put them in.

Good luck with the (eventual) upgrade!

**Reply by liuyanghu** [reply]
Do you mean the ". xrdp" connection profile of windows mstsc or Ubuntu user home profile? I'm very sorry to ask another question. Is there any recommended way to restore "lost" sessions if necessary? I sometimes try to retrieve the previously opened session through testing different port "5910", "5911", "5912" one by one。I found this way on Google and Stack Overflow。On stack overflow, it was also suggested that you can view the ports used by xrdp through some commands.

**Reply by matt335672** [reply]
Neither - I mean the file that mstsc.exe saves when you ask it to. It's just a text file, and you can set a display height and width in there directly. Setting these values for example lets me run a huge desktop with scrollbars on my small laptop:-

```
desktopwidth:i:3840
desktopheight:i:2160
```

The easiest way on your version of xrdp to see what sessions are running is to look for Xvnc X servers:-

<pre>
$ ps -ef | grep Xvnc
<b>testuser</b>   13390   13355  0 15:07 ?        00:00:00 Xvnc <b>:11</b> -auth .Xauthority -geometry <b>1364x768</b> -depth <b>32</b> -rfbauth /home/testuser/.vnc/sesman_passwd-testuser@xrdp-test:11 -bs -nolisten tcp -localhost -dpi 96
. . .
</pre>

That shows a session started by `testuser` on display number 11 with a screen size of 1364x768 and a colour depth of 32 bits. I'm using a much later version of xrdp than 0.6.1, but I think the same should apply.

**Comment by yardkeeper** [comment]
What I need to configure that I can to connect to same session ? Because its (Reconnect to an existing session) not working. RDP not VNC

**Comment by liuyanghu** [comment]
Follwing is answer from matt335672!
Befor this mail, I told him my xrdp verison is 0.6.
Hope that it can help you.

OK - that explains a lot. That's a very old version of xrdp. You must be running Ubuntu 16.04.

With that version of xrdp and sesman-Xvnc, the session is not resizeable. So when user A starts the session from his client, the session will have a certain width and height (say width = 1280, height = 768) and a certain colour depth (say 24 bits).

For user B to connect to the same session on this version, all of these three items must match. So, in this example, that means that user B will have to select a screen size of 1280x768, and a colour depth of 24 on the client parameters before connecting. Here is a screenshot showing what I mean:-

Does that make sense?

Later versions of xrdp support this a lot better, but even so, this method will work with them too.

| |
yanghu·liu
|
|
***@***.***
|
On 1/12/2022 ***@***.***> wrote：

What I need to configure that I can to connect to same session ? Because its (Reconnect to an existing session) not working

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
